src/libraries/System.Net.Http/src/System/Net/Http/Headers/AltSvcHeaderParser.cs

```diff
@@ -202,53 +202,36 @@ private static bool TryReadPercentEncodedAlpnProtocolName(string value, int star
             // Special-case expected values to avoid allocating one-off strings.
             switch (span.Length)
             {
-                case 2:
-                    if (span[0] == 'h')
+                case 2 when span is "h3":
+                    result = "h3";
+                    return true;
+
+                case 2 when span is "h2":
+                    result = "h2";
+                    return true;
+
+                case 3 when span is "h2c":
+                    result = "h2c";
+                    readLength = 3;
```

Comment:
```suggestion
```
Nit: existing redundant code

Reply:
CI passed so I merged. Feel free to delete it separately. I wasn't sure why this one case had that and I was afraid to delete it.